src/Runner.Worker/Handlers/CompositeActionHandler.cs

```diff
@@ -188,13 +171,11 @@ private async Task RunStepsAsync(List<IStep> compositeSteps)
                     }
                 }
 
-                step.ExecutionContext.ExpressionValues["env"] = envContext;
```

Comment:
this line just moved up

Reply:
At this point in the code, we're still in the middle of merging values into the env dictionary. So it makes more sense to move this statement up or down (i moved it up)